common/transport/amqp/devdoc/receiver_link_requirements.md

```diff
@@ -0,0 +1,104 @@
+# ReceiverLink Requirements
+
+## Overview
+
+The `ReceiverLink` class is internal to the SDK and shall be considered private. It shall not be used directly by clients of the SDK, who should instead rely on the higher-level constructs (`Client` classes provided by `azure-iot-device` and `azure-iothub`).
+
+The `ReceiverLink` class implements a state machine that manages the underlying `amqp10` link object used to receive messages from IoT Hub. It can be attached and detached manually, and will try to attach automatically if not already attached when setting up a message listener or trying to settle an existing message.
```

Comment:
>implements [](start = 25, length = 10)

Surfacing this information in the spec is nice I suppose, but actually doesn't really do anything for the spec.

While the Recevier Link utilizes a fsm that has zero affect on the api surface supplied.

Frankly I would remove any mention of it here.